qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/JobWorkerTest.java

```diff
@@ -56,81 +61,173 @@ public void shouldActivateJob() {
+    // given
+    client.createSingleJob(jobType, b -> {}, "{\"a\":1, \"b\":2, \"c\":3,\"d\":4}");
+
+    // when
+    final var fetchVariables = List.of("a", "b");
+    final var jobHandler = new RecordingJobHandler();
+    final var builder =
+        client
+            .getClient()
+            .newWorker()
+            .jobType(jobType)
+            .handler(jobHandler)
+            .fetchVariables(fetchVariables);
+    try (final var ignored = configurator.apply(jobType, builder)) {
+      Awaitility.await("until all jobs are activated")
+          .untilAsserted(() -> assertThat(jobHandler.getHandledJobs()).hasSize(1));
+    }
+
+    // then
+    final ActivatedJob job = jobHandler.getHandledJobs().get(0);
+    assertThat(job.getVariablesAsMap()).isEqualTo(Map.of("a", 1, "b", 2));
+  }
+
+  @Test
+  void shouldStreamAndActivateJobs() {
+    // given - a job created before any stream was registered, i.e. can only be polled
+    client.createSingleJob(jobType, b -> {});
+
+    // when - create a worker that streams, and create a new job after the stream is registered
+    final var jobHandler = new RecordingJobHandler();
+    final var builder =
+        client.getClient().newWorker().jobType(jobType).handler(jobHandler).enableStreaming();
+    try (final var ignored = builder.open()) {
+      awaitStreamRegistered(jobType);
+      client.createSingleJob(jobType, b -> {});
+
+      // then - expect both jobs to be activated
+      Awaitility.await("until all jobs are activated")
+          .untilAsserted(() -> assertThat(jobHandler.getHandledJobs()).hasSize(2));
+    }
+  }
+
   @Test
-  public void shouldFetchOnlySpecifiedVariables() {
+  void shouldRecreateStreamOnGatewayRestart() {
     // given
-    CLIENT_RULE.createSingleJob(jobType, b -> {}, "{\"a\":1, \"b\":2, \"c\":3,\"d\":4}");
+    final var jobHandler = new RecordingJobHandler();
+    final var builder =
+        client.getClient().newWorker().jobType(jobType).handler(jobHandler).enableStreaming();
 
     // when
-    final List<String> fetchVariables = List.of("a", "b");
+    try (final var ignored = builder.open()) {
+      awaitStreamRegistered(jobType);
+      // have to reuse the same gateway config, otherwise we need to recreate the client to point to
+      // the new gateway, which sort of negates the point of this test
+      final var gatewayConfig = CLUSTER.stopGateway();
+      // avoid flakiness by awaiting it gets removed properly; this may be removed once the
+      // following issue is finished: https://github.com/camunda/zeebe/issues/13389
+      awaitStreamRemoved(jobType);
+      CLUSTER.startGateway(gatewayConfig);
+      // need to stream being registered, as otherwise the job will be polled, not streamed
+      awaitStreamRegistered(jobType);
+      client.createSingleJob(jobType, b -> {});
+
+      // then - expect job to be activated
+      Awaitility.await("until all jobs are activated")
```

Comment:
While we could simply assert that the stream is recreated, since this is from a user PoV, users would want to make sure jobs are still activated after. It's a little brittle because we have to wait for the stream to be registered anyway before creating the job, but one day we may remove this requirement :sweat_smile: